### DISCUSSION TITLE
Error restoring container from checkpoint

### DISCUSSION THREAD

**Discussion by NikolaBo** [opencontainers/runc#3535] [Q&A] [answered]
I have a container running in runc, and I am able to checkpoint it: `runc checkpoint containername`. But when I go to restore it I get `ERRO[0000] JSON specification file config.json not found `. 

Any advice on where to start troubleshooting this? I know that my kernel supports CRIU. It seems odd that the dump/checkpoint succeeds but the restore complains about missing a config in the checkpoint. 

Version: 

> runc version 1.1.2
commit: a916309fff0f838eb94e928713dbc3c0d0ac7aa4
spec: 1.0.2-dev
go: go1.18.3
libseccomp: 2.5.3

**Comment by kolyshkin** [comment, marked as answer]
When you create a container, runc takes its configuration from OCI spec, which is in a file named `config.json` residing in either the current directory or the directory specified via the `--bundle` option.

When you restore a container, you are essentially re-creating it, so `config.json` (same as used to create this container) is required.

Hope that makes sense.

**Reply by NikolaBo** [reply]
Thanks!
